config/environments/production.rb

```diff
@@ -83,4 +83,42 @@
 
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
+
+  # Serve plugin static assets if the application is configured to do so
+  if config.serve_static_assets
+    ::Rails::Engine.subclasses.map(&:instance).each do |engine|
+      if File.exist?("#{engine.root}/public/assets")
+        app.middleware.use ::ActionDispatch::Static, "#{engine.root}/public"
```

Comment:
This belongs in the environment rather than in the initializer, as `app.middleware` isn't modifiable outside of this context.